driver-core/src/main/com/mongodb/ConnectionString.java

```diff
@@ -298,7 +301,7 @@ public ConnectionString(final String uri) {
         WRITE_CONCERN_KEYS.add("w");
         WRITE_CONCERN_KEYS.add("wtimeoutms");
         WRITE_CONCERN_KEYS.add("fsync");
-        WRITE_CONCERN_KEYS.add("j");
+        WRITE_CONCERN_KEYS.add("journal");
```

Comment:
This is a backwards compatibility break.  We have to support both options and give journal precedence over j.